fontforge/sfd.c

```diff
@@ -3631,26 +3658,29 @@ static void rle2image(struct enc85 *dec,int rlelen,struct _GImage *base) {
     }
 }
 
-#ifndef _NO_LIBPNG
 
-enum MIME { UNKNOWN, PNG }; // We only understand PNG for now.
+enum Image2MIME { IM2_UNKNOWN, IM2_PNG, IM2_REFERENCE };
 
-enum MIME SFDGetImage2MIME(FILE *sfd) {
+enum Image2MIME SFDGetImage2MIME(FILE *sfd) {
     char mime[128];
 
     if ( !getname(sfd, mime) ) {
         IError("Failed to get a MIME type, file corrupt");
-        return UNKNOWN;
+        return IM2_UNKNOWN;
+    }
+
+    if ( strmatch(mime, "image/png")==0 ) {
+        return IM2_PNG;
     }
 
-    if ( !(strmatch(mime, "image/png")==0) ) {
-        IError("MIME type received—%s—is not recognized", mime);
-        return UNKNOWN;
+    if ( strmatch(mime, "reference")==0 ) {
```

Comment:
You may already have thought about this, but, it seems to me that "text/uri-list" might work just as well for the immediate use case and also allow us to comply with the MIME specification.

Reply:
Well... Remote files aren't supported, and probably shouldn't be, so it's not truly a URI...Also I don't think a file URI can be relative

Reply:
It's okay to reject an unsupported protocol in a URI. And relative paths are easy. If the file is reached via the same protocol (as would always be the case here), you just write the relative path the same way you would for a relative UNIX path.

Reply:
Oh, so would I give  it a protocol other than `file://`?

Reply:
Those were separate points.

On the first point, we don't need to worry about remote files. If there is an http or ftp path, we just warn and ignore it.

On the second point, a relative path would not explicitly list the protocol, just like you can write `<img src="images/logo.png"/>`.

Reply:
For the purposes of MIME, is `images/logo.png` a valid URI? Would `images\logo.png` be?

Reply:
We ought to be no more permissive than web standards, so certainly no backslashes.

Reply:
Sorry, I think `text/uri-list` is always supposed to have the protocol, [according to Mike Amundsen](http://amundsen.com/hypermedia/urilist/).

Likewise, [RFC 2483 &sect; 5](https://tools.ietf.org/html/rfc2483#section-5) states:

>     The format of text/uri-list resources is:
> 
>     1) Any lines beginning with the '#' character are comment lines
>         and are ignored during processing. (Note that URIs may contain
>         the '#' character, so it is only a comment character when it is
>         the first character on a line.)
> 
>     2) The remaining non-comment lines shall be URIs (URNs or URLs),
>         encoded according to the URL or URN specifications (RFC2141,
>         RFC1738 and RFC2396). Each URI shall appear on one and only one
>         line. Very long URIs are not broken in the text/uri-list format.
>         Content-transfer-encodings may be used to enforce line length
>         limitations.

The HTML standard is more permissive...it supports both URIs and relative paths, but a relative path is not itself a URI, right? Just as it makes no sense to have a relative `http://` URI it makes no sense to have a relative `file://` URI. A URI must always state a protocol as well.

I don't mind breaking the standard, but please be aware of this.

Reply:
It seems ambiguous enough that we can claim to have made a good faith effort. If we flood the Internet with enough SFD files with relative paths, they'll eventually resolve the ambiguity in our favor.